src/System.Transactions/tests/NonMsdtcPromoterTests.cs

```diff
@@ -2337,23 +1989,14 @@ public static void TestCase_SetDistributedIdAtWrongTime()
                     try
                     {
                         SetDistributedTransactionId(dummyPSPE, Transaction.Current, guidToSet);
-                        TestFailed(testCaseDescription, failureDetail);
+                        bool shouldNotBeExecuted = true;
+                        Assert.False(shouldNotBeExecuted);
                     }
-                    catch (TransactionException ex)
+                    catch (Exception ex)
                     {
+                        Assert.True(ex is TransactionException || (ex is TargetInvocationException && ex.InnerException is TransactionException));
```

Comment:
Nit: this could be done with Assert.ThrowsAny, e.g.
```C#
Exception ex = Assert.ThrowsAny<Exception>(() => SetDistributedTransactionId(dummyPSPE, Transaction.Current, guidToSet));
Assert.True(ex is TransactionException || ex is TargetInvocationException && ex.InnerException is TransactionException);
```